arch/arm64/src/common/arm64_syscall.c

```diff
@@ -217,10 +219,13 @@ uint64_t *arm64_syscall(uint64_t *regs)
 
       case SYS_switch_context:
         {
+          struct tcb_s *rtcb = regs[REG_X1];
+          struct tcb_s *tcb = regs[REG_X2];
+
           DEBUGASSERT(regs[REG_X1] != 0 && regs[REG_X2] != 0);
-          *(uint64_t **)regs[REG_X1] = regs;
+          *(uint64_t **)(&rtcb->xcp.regs) = regs;
```

Comment:
Isn't this redundant now, @hujun260 ?

Reply:
If not, `rtcb->xcp.regs = regs` should suffice

Reply:
done